### PR TITLE
Switch CQRS sample to use sharded daemons

### PR DESCRIPTION
**Pull request by chbatey**
Well that was easy

### CODE REVIEW

**Comment by johanandren** [comment]
😈

File: akka-sample-cqrs-java/src/main/java/sample/cqrs/EventProcessor.java [hunk 1]
```diff
@@ -4,47 +4,38 @@
...
-
-    ClusterSharding.get(system).init(Entity.of(eventProcessorEntityKey, entityContext ->
-      EventProcessor.create(eventProcessorStream.apply(entityContext.getEntityId()))).withRole("read-model"));
+    ShardedDaemonProcessSettings shardedDaemonSettings = ShardedDaemonProcessSettings.create(system)
+            .withKeepAliveInterval(settings.keepAliveInterval)
+            .withShardingSettings(ClusterShardingSettings.create(system).withRole("read-model"));
 
-    KeepAlive.init(system, eventProcessorEntityKey);
+    ShardedDaemonProcess.get(system)
+            .init(Void.class, "event-processors-" + settings.id, settings.parallelism, i -> EventProcessor.create(eventProcessorStream.apply(settings.tagPrefix + "-" + i)), shardedDaemonSettings, Optional.empty());
```
Comment (patriknw) [inline akka-sample-cqrs-java/src/main/java/sample/cqrs/EventProcessor.java]: that's a very long line

File: akka-sample-cqrs-java/src/main/resources/logback.xml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <configuration>
     <appender name="STDOUT" target="System.out" class="ch.qos.logback.core.ConsoleAppender">
         <encoder>
-            <pattern>[%date{ISO8601}] [%level] [%logger] [%marker] [%thread] - %msg {%mdc}%n</pattern>
+            <pattern>[%date{ISO8601}] [%level] [%logger] [{%mdc}] [%marker] [%thread] - %msg %n</pattern>
```
Comment (patriknw) [inline akka-sample-cqrs-java/src/main/resources/logback.xml]: `%mdc` can be rather much, so better to keep it at then end

File: akka-sample-cqrs-scala/src/main/resources/logback.xml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 <configuration>
     <appender name="STDOUT" target="System.out" class="ch.qos.logback.core.ConsoleAppender">
         <encoder>
-            <pattern>[%date{ISO8601}] [%level] [%logger] [%marker] [%thread] - %msg {%mdc}%n</pattern>
+            <pattern>[%date{ISO8601}] [%level] [%logger] [%marker] [{%mdc}] [%thread] - %msg %n</pattern>
```
Comment (patriknw) [inline akka-sample-cqrs-scala/src/main/resources/logback.xml]: ^

File: akka-sample-cqrs-scala/src/main/scala/sample/cqrs/EventProcessor.scala [hunk 4]
```diff
@@ -35,41 +33,30 @@ import org.slf4j.LoggerFactory
...
-
-    ClusterSharding(system).init(Entity(eventProcessorEntityKey)(entityContext =>
-      EventProcessor(eventProcessorStream(entityContext.entityId))).withRole("read-model"))
-
-    KeepAlive.init(system, eventProcessorEntityKey)
+      val shardedDaemonSettings = ShardedDaemonProcessSettings(system)
+          .withKeepAliveInterval(settings.keepAliveInterval)
+          .withShardingSettings(ClusterShardingSettings(system).withRole("read-model"))
+      ShardedDaemonProcess(system)
+        .init[Nothing](s"event-processors-${settings.id}", settings.parallelism, i => EventProcessor(eventProcessorStream(s"${settings.tagPrefix}-$i")), shardedDaemonSettings, None)
```
Comment (patriknw) [inline akka-sample-cqrs-scala/src/main/scala/sample/cqrs/EventProcessor.scala]: long line

File: akka-sample-cqrs-scala/src/main/scala/sample/cqrs/EventProcessor.scala [hunk 5]
```diff
@@ -35,41 +33,30 @@ import org.slf4j.LoggerFactory
...
   }
 
-  def apply(eventProcessorStream: EventProcessorStream[_]): Behavior[Ping.type] = {
-
-    Behaviors.setup { context =>
+  def apply(eventProcessorStream: EventProcessorStream[_]): Behavior[Nothing] = {
+    Behaviors.setup[Nothing] { ctx  =>
+      ctx.log.info("Event processor running {}", eventProcessorStream)
```
Comment (patriknw) [inline akka-sample-cqrs-scala/src/main/scala/sample/cqrs/EventProcessor.scala]: toString of the `eventProcessorStream` is probably not that pretty?

File: akka-sample-cqrs-java/src/main/resources/logback.xml (hunk 2 above)
Comment (chbatey) [reply akka-sample-cqrs-java/src/main/resources/logback.xml]: okay. I changed as for empty mdc it looks like an extra marker

File: akka-sample-cqrs-scala/src/main/scala/sample/cqrs/EventProcessor.scala (hunk 5 above)
Comment (chbatey) [reply akka-sample-cqrs-scala/src/main/scala/sample/cqrs/EventProcessor.scala]: 👍

**Comment by patriknw** [comment]
I don't think the latest changes were pushed. I'll follow up on that because I have some other updates to this sample.

**Pull request merged**